lxd/instance/drivers/driver_qemu.go

```diff
@@ -7193,10 +7227,17 @@ func (d *qemu) MigrateReceive(args instance.MigrateReceiveArgs) error {
 						return fmt.Errorf("Failed creating instance snapshot record %q: %w", snapArgs.Name, err)
 					}
 
+					snapInstOps = append(snapInstOps, snapInstOp)
+					snapCreationErrs = append(snapCreationErrs, err)
 					revert.Add(cleanup)
-					defer snapInstOp.Done(err)
 				}
 			}
+
+			defer func() {
```

Comment:
again, im not sure what the linter is asking you to fix, but this change is actually going to cause operations to be left open if there is an earlier return.

Reply:
Same aanswer as above. I can look for a kind of `//no-lint`  message to put there to avoid the issue.

Reply:
Yeah im interested in the why it was thinking a defer in the for loop was a problem before we stick a no-lint in there.

Reply:
I can do what's explained here instead: https://stackoverflow.com/questions/45617758/proper-way-to-release-resources-with-defer-in-a-loop#:~:text=The%20whole%20point,without%20defer%3A